scilpy/tractanalysis/reproducibility_measures.py

```diff
@@ -124,6 +123,59 @@ def get_endpoints_density_map(streamlines, dimensions, point_to_select=1):
     return endpoints_map
 
 
+def get_head_tail_density_maps(streamlines, dimensions):
+    """
+    Compute two separate endpoints density maps for the head and tail
+    Parameters
+    ----------
+    streamlines: list of ndarray
+        The list of streamlines to compute endpoints density from.
+    dimensions: tuple
+        The shape of the reference volume for the streamlines.
+
+    Returns
+    -------
+    A tuple containing
+        ndarray: A ndarray where voxel values represent the density of
+            head endpoints.
+        ndarray: A ndarray where voxel values represent the density of
+            tail endpoints.
+    """
+    endpoints_map_head = np.zeros(dimensions)
+    endpoints_map_tail = np.zeros(dimensions)
+
+    for streamline in streamlines:
+        xyz = streamline[0, :].astype(int)
+        endpoints_map_head[xyz[0], xyz[1], xyz[2]] += 1
+
+        xyz = streamline[-1, :].astype(int)
+        endpoints_map_tail[xyz[0], xyz[1], xyz[2]] += 1
+
+    return endpoints_map_head, endpoints_map_tail
+
+
+def approximate_surface_node(roi):
```

Comment:
This is shown as new code, is this a copy paste from elsewhere?
Did I sent you code or what it already in scilpy?

EDIT: Oh no it was in a script, forget it

Reply:
Yes, it was originally in scripts/scil_normalize_connectivity.py. I have extracted this method from that script and added the method to scilpy/tractanalysis/reproducibility_measures.py.